project-koku/koku
# Copy ocp-on-aws, ocp-on-azure data into partitioned tables.

## Description

This is a multi-process job that will fire off 4 worker processes. Each process will connect to the database read from a queue to get the schema and table information to perform the copy. So there should be up to 4 tables from up to 2 schemata being copied concurrently.

## Testing

The script can be run directly as a test. 
1. Re-init your database making sure that `reporting_ocpawscostentrylineitem_*daily_summary_p` tables have been created
2. Load some customer data that will populate those base tables (same table names, but not ending in `_p`)
3. Execute `./scripts/copy_ocp_aws_azure_data.py <num_processes>` (I'd suggest at least `2` for this value on localdev)
4. You will see logs for the main process as well as each worker created. These logs will show the tables being copied and the number of records copied once complete.
5. Once all worker processes have completed, the main process will exit.
